### ISSUE TITLE
Bug in Angular 1.2.28? Using ng-options on a HTML select tag, produces not selectable last option element.

### ISSUE THREAD

**Issue by roelvanlisdonk** [angular/angular.js#10669]
If have created a plunker to demonstrate the problem:
http://plnkr.co/edit/ndTQmTftKcwn7emBx84w?p=preview

First: 
Create a HTML select tag, like:

<code>
&lt;select ng-model="module.reason" ng-options="reason for reason in resources.reasons" required&gt;&lt;/select&gt;
</code>

Create a controller like:
<code>
            var controller = function ($scope)
            {
                $scope.title = "";

                $scope.module = {
                    reason: null
                };

                $scope.resources = {
                    reasons: [
                        "Reason 1",
                        "Reason 2",
                        "Reason 3",
                        "Reason 4"
                    ]
                };
            };
</code>

Now when you run the page, immediately select the last option (in this case "Reason 4").

No option will be selected, the select tag will be empty, when you select an other option like "Reason 2" and then select the "Reason 4" option, it will be selected, but  when you change the selection back to the empty selection and then select the "Reason 4" option, it will be empty again.

**Comment by frfancha** [comment]
Well very strange.
And you can reproduce the problem without jQuery, with the ng-required, without the style...
With this you have the problem:
`    <select ng-model="module_string"
            ng-options="reason as reason for reason in reasons_string"
            ></select>
$scope.reasons_string = [
                        "R1",
                        "R2"
                ];`

I was wondering why we don't have it (we use angular 1.2 a lot!!), and in fact you don't have the problem if you do this:
`<select ng-model="module_object"
            ng-options="reason.V as reason.D for reason in reasons_object"
            ></select>`
and
`$scope.reasons_object = [
                        {V:1, D:"R1"},
                        {V:2, D:"R2"}
                ];`

**Comment by roelvanlisdonk** [comment]
Even when I remove jQuery and all styles, just like the html below, I still have the problem.
But you are right, when I use objects, instead of strings, the problem does not exist.
But I think it should work with strings too, don't you think?

<code>
&lt;!DOCTYPE html&gt;
&lt;html xmlns:ng="http://angularjs.org" id="ng-app" ng-app="spike"&gt;
&lt;head&gt;
    &lt;title&gt;Spike page&lt;/title&gt;
    &lt;meta charset="utf-8"&gt;
    &lt;meta http-equiv="X-UA-Compatible" content="IE=edge"&gt;
&lt;/head&gt;
&lt;body ng-controller="main"&gt;
    &lt;select ng-model="module.reason"
            ng-options="reason for reason in resources.reasons"
            required&gt;&lt;/select&gt;

    &lt;script src="//ajax.googleapis.com/ajax/libs/angularjs/1.2.28/angular.min.js"&gt;&lt;/script&gt;

    &lt;!-- App scripts --&gt;
    &lt;script&gt;

        (function ()
        {
            "use strict";

            var app = angular.module("spike", []);
        }());

        (function ()
        {
            "use strict";

            var app = angular.module("spike");

            var controller = function ($scope)
            {
                $scope.title = "";

                $scope.module = {
                    reason: null
                };

                $scope.resources = {
                    reasons: [
                        "Reason 1",
                        "Reason 2",
                        "Reason 3",
                        "Reason 4"
                    ]
                };
            };

            app.controller("main", ["$scope", controller]);
        }());
    &lt;/script&gt;
&lt;/body&gt;
&lt;/html&gt;
</code>

**Comment by dhwang** [comment]
workaround: Add a default empty option in like <option value=""> </option>
and the problem go away

**Comment by shadowmaru** [comment]
My case is somewhat different, but similar. I have a select with a list of strings (or numbers) and an ng-model that was loaded from an object. If the ng-model is the last option of the list, it shows the empty selection, although the ng-model is correctly set.

The workaround by @dhwang fixed the problem.

**Comment by craigbroadman** [comment]
Using the plunker in the original post it seems that this bug first manifested in version 1.2.26 - https://github.com/angular/angular.js/commit/11d2242df65b2ade0dabe366a0c42963b6d37df5

**Comment by cuipengfei** [comment]
@dhwang thanks, this workaround worked for me

**Comment by suzanshakya** [comment]
@dhwang Your fix makes last option selectable, but also makes first option not selectable.

**Comment by WayneFurmidge** [comment]
After a little digging this looks to have been a problem since 1.2.26 (working in 1.3.0)

The code that removes the un-needed last element was updated to call

selectCtrl.removeOption(option.label);

Which is causing the rendering of an unknown option. Removing that line of code dose fix things, but I cant tell why it was added between 1.2.25 and 1.2.26 so i am resistant to submit a pull request

**Comment by dhwang** [comment]
huh.. I'm using 1.2.23 and it has this problem that forced me to add a default empty option.

**Comment by thierrylaz** [comment]
Hi,
I also have a similar case.
if I use the script $ ("select"). selecter () to my style, all values in my "optionSelect" are empty set, and if I comment the script, I get the values.
Are you a solution to this conflict?

**Comment by Narretz** [comment]
1.2.x only receives critical bug fixes; closing
